### DISCUSSION TITLE
Using anchors in mathematical formulas

### DISCUSSION THREAD

**Discussion by psads-git** [cetz-package/cetz#521] [Q&A]
I need to set anchors in some letters of a mathematical formula to draw arrows from those anchors. The final plot I am aiming to is attached (the arrow tips are missing). Could you please help me? Thanks!

![image](https://github.com/johannes-wolf/cetz/assets/75945439/0f4effab-c24b-40e9-916c-ad3fe7d1f309)

**Comment by johannes-wolf** [comment]
This is not possible right now. The problem is, that Typst `locate` function must return content, but to make this work, cetz has to somehow modify its context object from the locate callback, which is not possible.

You may take a look at the [pinit](https://github.com/typst/packages/tree/main/packages/preview/pinit/0.1.3) package.

**Comment by psads-git** [comment]
Thanks, @johannes-wolf, for your answer and for suggesting [pinit](https://github.com/typst/packages/tree/main/packages/preview/pinit/0.1.3) package!
